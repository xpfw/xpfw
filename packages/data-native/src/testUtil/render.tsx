import * as React from "react"
import rtr from "react-test-renderer"

export default (Component: any, name: string) => {
  const rendered = rtr.create(Component)
  expect(rendered.toJSON()).toMatchSnapshot(name)
  rendered.unmount()
}
