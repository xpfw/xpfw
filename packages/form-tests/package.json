{
  "name": "@xpfw/form-tests",
  "version": "1.0.0-beta.12",
  "main": "dist",
  "typings": "dist",
  "license": "MIT",
  "peerDependencies": {
    "lodash-es": "^4.x.x",
    "moment": "2.x.x",
    "react": "^16.x.x",
    "@xpfw/form": "1.x.x"
  },
  "dependencies": {
  },
  "scripts": {
    "build": "./node_modules/.bin/rollup -c",
    "dev": "./node_modules/.bin/rollup -c -w",
    "test": "./node_modules/.bin/jest --ci",
    "update": "ncu -a",
    "prepare": "npm run build"
  },
  "devDependencies": {
    "@types/jest": "^24.0.4",
    "@types/lodash": "^4.14.121",
    "@types/lodash-es": "^4.17.2",
    "@types/mockdate": "^2.0.0",
    "@types/react": "^16.8.3",
    "@types/react-test-renderer": "^16.8.1",
    "@types/socket.io-client": "^1.4.32",
    "lodash-es": "^4.17.11",
    "moment": "^2.24.0",
    "@xpfw/form": "^1.0.0-beta.12",
    "jest": "^24.1.0",
    "jest-cli": "^24.1.0",
    "mockdate": "^2.0.2",
    "react": "^16.8.1",
    "react-test-renderer": "^16.8.2",
    "rollup": "^1.1.2",
    "rollup-plugin-typescript2": "^0.19.2",
    "rollup-plugin-uglify": "^6.0.2",
    "ts-jest": "=23.10.5",
    "tslib": "^1.9.3",
    "typescript": "^3.3.3",
    "uglify-es": "^3.3.10"
  },
  "jest": {
    "globals": {
      "ts-jest": {
        "diagnostics": false
      }
    },
    "transform": {
      ".(ts|tsx)": "ts-jest"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(ts|tsx|js)$",
    "testPathIgnorePatterns": [
      "\\.snap$",
      "<rootDir>/node_modules/",
      "<rootDir>/lib",
      "<rootDir>/dist"
    ],
    "moduleNameMapper": {
      "preact-render-to-json": "preact-render-to-json",
      "preact": "preact",
      "react": "preact-compat",
      "react-dom": "preact-compat"
    },
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "testURL": "http://localhost/"
  }
}
