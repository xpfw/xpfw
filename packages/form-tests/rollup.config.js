// rollup.config.js
import typescript from "rollup-plugin-typescript2"
import {uglify} from "rollup-plugin-uglify"
import { minify } from "uglify-es"

export default {
  input: 'src/index.ts',
  output: {
    file: 'dist/index.js',
    format: 'cjs'
  },
  plugins: [
    typescript(),
    // uglify({}, minify)
  ],
  external: [
    "lodash",
    "moment",
    "mockdate",
    "feathers-client",
    "resub",
    "resub-persist",
    "socket.io-client",
    "@xpfw/validate",
    "@xpfw/form-shared",
    "react",
    "react-dom",
    "preact-render-to-json"
  ]
};