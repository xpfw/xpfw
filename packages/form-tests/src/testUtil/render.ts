import * as React from "react"
import * as render from "react-test-renderer"

const renderSnapshot = (component: any, msg: string) => {
  const tree = render.create(component)
  expect(tree.toJSON()).toMatchSnapshot(msg)
  tree.unmount()
}

export default renderSnapshot
