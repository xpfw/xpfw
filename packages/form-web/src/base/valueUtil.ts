import { get } from "lodash"

const setFromEvent = (setValue: any, eventKey: string) => {
  return (e: any) => {
    setValue(get(e, eventKey))
  }
}

export { setFromEvent }
