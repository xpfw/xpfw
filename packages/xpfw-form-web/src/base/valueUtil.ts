import { get } from "lodash"

const setFromEvent = (thisRef: {props: {setValue: any}}, eventKey: string) => {
  return (e: any) => {
    thisRef.props.setValue(get(e, eventKey))
  }
}

export { setFromEvent }
