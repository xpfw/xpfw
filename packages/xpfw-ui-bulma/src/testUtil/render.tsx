import * as rtr from "preact-render-to-json"
import * as React from "react"

export default (Component: any, name: string) => {
  expect(rtr(Component)).toMatchSnapshot(name)
}
