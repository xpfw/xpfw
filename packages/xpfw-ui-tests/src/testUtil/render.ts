import * as rtr from "preact-render-to-json"
import * as React from "react"
const r: any = rtr
export default (Component: any, name: string) => {
  expect(r(Component)).toMatchSnapshot(name)
}
