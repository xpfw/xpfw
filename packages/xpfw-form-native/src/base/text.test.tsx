import { ComponentRegistry, FormStore, SharedField } from "@xpfw/form-shared"
import { tests } from "@xpfw/form-tests"
import {  FieldType, IField } from "@xpfw/validate"
import { set } from "lodash"
import * as MockDate from "mockdate"
import * as React from "react"
import render from "../testUtil/render"
import TextField, { setDate } from  "./text"
import { setFromEvent } from "./valueUtil"
MockDate.set(new Date(2011, 4, 5, 16, 20, 42, 1337))

ComponentRegistry.registerComponent(FieldType.Text, TextField)
ComponentRegistry.registerComponent(FieldType.Number, TextField)
ComponentRegistry.registerComponent(FieldType.Date, TextField)
ComponentRegistry.registerComponent(FieldType.Password, TextField)
ComponentRegistry.registerComponent(FieldType.Slider, TextField)

test("Text Field Test", tests.text)
