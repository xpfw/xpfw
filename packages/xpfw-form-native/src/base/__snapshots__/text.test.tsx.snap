// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Field Test: as number field 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      myNum
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "myNum",
          "type": 3,
          "validate": Object {},
        }
      }
      keyboardType="numeric"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: as number field with number 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      myNum
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "myNum",
          "type": 3,
          "validate": Object {},
        }
      }
      keyboardType="numeric"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value={5432}
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: as number field with string 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      myNum
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "myNum",
          "type": 3,
          "validate": Object {},
        }
      }
      keyboardType="numeric"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="gfsdfasd"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: as number field with string number 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      myNum
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "myNum",
          "type": 3,
          "validate": Object {},
        }
      }
      keyboardType="numeric"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="125"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: as slider field 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      myNum
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "myNum",
          "type": 15,
          "validate": Object {},
        }
      }
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: as slider field with number 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      myNum
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "myNum",
          "type": 15,
          "validate": Object {},
        }
      }
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value={5432}
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: as slider field with string 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      myNum
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "myNum",
          "type": 15,
          "validate": Object {},
        }
      }
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="gfsdfasd"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: as slider field with string number 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      myNum
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "myNum",
          "type": 15,
          "validate": Object {},
        }
      }
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="125"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: date with content 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      dateVal
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "dateVal",
          "type": 5,
          "validate": Object {
            "type": 2,
          },
        }
      }
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="2012-01-11"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: datetime-local with content 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      dateVal
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "dateVal",
          "type": 5,
          "validate": Object {
            "type": 2,
          },
        }
      }
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="2018-02-22T07:55"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: empty datetime-local field 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      dateVal
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "dateVal",
          "type": 5,
          "validate": Object {
            "type": 2,
          },
        }
      }
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: empty render 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={null}
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: namefield 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      guestName
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "guestName",
          "type": 2,
          "validate": Object {
            "permission": Object {
              "create": "guest",
              "find": "guest",
              "remove": "guest",
              "update": "guest",
            },
          },
        }
      }
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: namefield changed va 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      guestName
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "guestName",
          "type": 2,
          "validate": Object {
            "permission": Object {
              "create": "guest",
              "find": "guest",
              "remove": "guest",
              "update": "guest",
            },
          },
        }
      }
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="myval"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: password field 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      pwval
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "pwval",
          "type": 7,
          "validate": Object {
            "type": 2,
          },
        }
      }
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={true}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View />
</View>
`;

exports[`Text Field Test: time with content 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      dateVal
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      field={
        Object {
          "mapTo": "dateVal",
          "type": 5,
          "validate": Object {
            "type": 2,
          },
        }
      }
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={false}
      setValue={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="16:20"
    />
  </View>
  <View />
</View>
`;
